File: video-service-client/src/main/java/com/boclips/videos/service/client/internal/resources/CollectionsResource.java
package com.boclips.videos.service.client.internal.resources;

import lombok.Data;

import java.util.List;

@Data
public class CollectionsResource {
    private EmbeddedCollectionsResource _embedded;

    public List<CollectionResource> getCollections() {
        return this._embedded.getCollections();
    }
}

@Data
class EmbeddedCollectionsResource {
    private List<CollectionResource> collections;
}

File: video-service-api/src/main/kotlin/com/boclips/videos/api/response/collection/CollectionsResource.kt
package com.boclips.videos.api.response.collection

import com.boclips.videos.api.response.HateoasLink
import com.fasterxml.jackson.annotation.JsonInclude
import org.springframework.hateoas.PagedModel

class CollectionsResource(
    var _embedded: CollectionsWrapperResource,
    @JsonInclude(JsonInclude.Include.NON_NULL)
    var page: PagedModel.PageMetadata? = null,
    @JsonInclude(JsonInclude.Include.NON_NULL)
    var _links: Map<String, HateoasLink>? = null
)

data class CollectionsWrapperResource(
    val collections: List<CollectionResource>
)
